### PR TITLE
Prepare tune jb build

### PR DESCRIPTION
**Pull request by akosyakov**
## Description

It is supporting PR for https://github.com/gitpod-io/gitpod/pull/12390

This PR introduces a new workspace layer to deliver JB plugin independent from JB backend. It would allow to reduce build times from 15mins to 5mins on changes in JB backend plugin, since it does not require rebuild of all variants of JB backend images.

## Related Issue(s)

Related https://github.com/gitpod-io/gitpod/issues/12268

### Backward compatibility

Test startup of desktop IDEs in this prev env from dashboard and JB GW.

### Optimisation

Test https://github.com/gitpod-io/gitpod/pull/12390

## Release Notes

```release-note
NONE
```

## Werft options:

- [x] /werft with-preview

### CODE REVIEW

**Comment by roboquat** [comment]
@akosyakov: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by akosyakov** [comment]
@andreafalzetti @felladrin Could you help to test for regressions? 🙏

**Comment by akosyakov** [comment]
/hold

IDE team should check for regressions

**Comment by andreafalzetti** [comment]
Tested the workspace start with IntelliJ from prev env and works. The code changes look good but difficult to follow without knoledge around registry and ws-manager. To make easier to review and boos knowledge sharing, in the future we can consider adding a super quick loom explaining the changes for these sort of PRs 😄

**Comment by akosyakov** [comment]
> Do you mean starting a workspace from JB GW for prev env? Is it possible? How?

@andreafalzetti You can configure Gitpod GW Plugin to use prev env host: https://www.gitpod.io/docs/ides-and-editors/jetbrains-gateway#configuring-the-jetbrains-gateway-host

**Review by svenefftinge — APPROVED** [review]

**Comment by svenefftinge** [comment]
Code looks good, but the server's IDE_CONFIG doesn't have the property. So couldn't test.

**Comment by akosyakov** [comment]
@svenefftinge yes, this PR only too enable tuning, new configuration in https://github.com/gitpod-io/gitpod/pull/12390 but we can switch only all workspaces running new ws-manager

**Review by andreafalzetti — APPROVED** [review]

File: components/registry-facade-api/imagespec.proto [hunk 1]
```diff
@@ -16,6 +16,8 @@ message ImageSpec {
     string desktop_ide_ref = 4;
     // supervisor_ref points to an image denotign the supervisor to use
     string supervisor_ref = 5;
+    // desktop_ide_plugin_ref points to an image denotign the desktop IDE plugin to use
```
Comment (sagor999) [inline components/registry-facade-api/imagespec.proto:19]:
```suggestion
    // desktop_ide_plugin_ref points to an image denoting the desktop IDE plugin to use
```

File: components/registry-facade-api/imagespec.proto [hunk 2]
```diff
@@ -16,6 +16,8 @@ message ImageSpec {
     string desktop_ide_ref = 4;
     // supervisor_ref points to an image denotign the supervisor to use
```
Comment (sagor999) [inline components/registry-facade-api/imagespec.proto:17]:
```suggestion
    // supervisor_ref points to an image denoting the supervisor to use
```

**Review by sagor999 — APPROVED** [review]

**Comment by akosyakov** [comment]
@aledbf @csweichel i think it needs your approves as well since we are changing api of ws-manager

File: components/ws-manager-api/core.proto [hunk 3]
```diff
@@ -318,6 +318,8 @@ message IDEImage {
     string desktop_ref = 2;
     // supervisor_ref is a reference to an OCI image used as supervisor
     string supervisor_ref = 3;
+    // desktop_plugin_ref is an optional reference to an OCI image used for serving desktop IDE plugin
+    string desktop_plugin_ref = 4;
```
Comment (akosyakov) [inline components/ws-manager-api/core.proto:322]: @csweichel @aledbf 

Maybe it is better if we switch to just `refs` to make it easier to change IDE layers in backward compatible way without relying on deployments of ws-manager.

**Comment by felladrin** [comment]
ℹ️ I'm connected to the Preview Environment to review it.

**Review by felladrin — APPROVED** [review]
The code is clear. And I've tested different repositories on all 4 JB IDEs, alternating between Stable and EAP versions.

File: components/ws-manager-api/core.proto (hunk 3 above)
Comment (csweichel) [reply components/ws-manager-api/core.proto:322]: How often do we expect these layers to change down the road?

Comment (akosyakov) [reply components/ws-manager-api/core.proto:322]: We maybe would like to make some changes for VS Code optimisation too.

The thing is that I'm not sure why it should be specific here, ws-manager does not seem to care about the semantic.  Semantic is defined by the IDE service. ws-manager just need to apply them. Maybe we should rather then remove this dependency.

**Review by csweichel — APPROVED** [review]

File: components/ws-manager-api/core.proto (hunk 3 above)
Comment (akosyakov) [reply components/ws-manager-api/core.proto:322]: We decided first to deal with extracting IDE service from Server.

**Pull request merged**